lib/restAPIs/objectManagement.js

```diff
@@ -0,0 +1,83 @@
+const assert = require('assert');
+
+const assertParam = require('../util/assertParam');
+const RestUtil = require('../util/RestUtil');
+const buildCommonQuery = require('../util/buildCommonQueryParams');
+
+module.exports = class RestTopic extends RestUtil {
+  _getBase(){
+    return 'objects';
+  }
+
+  getObject(objectId, type, fields){
+    assert(objectId, 'No objectId provided');
+    assert(type, 'No type provided');
+    
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.type = type;
+
+    return this._makeRequest(`${this._getBase()}/${objectId}`, queryParameter, 'GET', this.getProjectHeader())
+    .then(result => this.throwIfFailed(result, 200));
+  }
+
+  updateObject(objectId, type, objectsBody, flags, fields){
+    assert(objectId, 'No objectId provided');
+    assert(type, 'No type provided');
+    assert(objectsBody, 'No objectsBody provided');
+
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.type = type;
+    if (flags) {
+      queryParameter.flags = flags;
+    }
+    return this._makeRequest(`${this._getBase()}/${objectId}`, objectsBody, 'PUT', this.getProjectHeader(), queryParameter)
+    .then(result => this.throwIfFailed(result, 200));
+  }
+
+  copyObject(objectId, type, objectsBody, fields){
+    assert(objectId, 'No objectId provided');
+    assert(type, 'No type provided');
+    assert(objectsBody, 'No objectsBody provided');
+
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.type = type;
+
+    return this._makeRequest(`${this._getBase()}/${objectId}/copy`, objectsBody, 'POST', this.getProjectHeader(), queryParameter)
+    .then(result => this.throwIfFailed(result, 201));
+  }
+
+  deleteObject(objectId, type){
+    assert(objectId, 'No objectId provided');
+    assert(type, 'No type provided');
+
+    const queryParameter = {
+      type: type
+    };
+
+    return this._makeRequest(`${this._getBase()}/${objectId}`, {}, 'DELETE', this.getProjectHeader(), queryParameter)
+    .then(result => this.throwIfFailed(result, 204));
+  }
+
+  certifyObject(objectId, type, certify){
+    assert(objectId, 'No objectId provided');
+    assert(type, 'No type provided');
+    assert(certify, 'No certify value provided');
+
+    const queryParameter = {
+      type: type,
+      certify: certify
+    };
+
+    return this._makeRequest(`${this._getBase()}/${objectId}/certify`, {}, 'PUT', this.getProjectHeader(), queryParameter)
+    .then(result => this.throwIfFailed(result, 200));
+  }
+
+  recommendObject(objectId, offset, limit, fields){
```

Comment:
I think this should be `getRecommendations`

Reply:
https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Object%20Management/getRecommendations

Reply:
The curl URL when you test it on the page is:

`https://demo.microstrategy.com/MicroStrategyLibrary/api/objects/12345/**recommendations**?offset=0&limit=-1`

which just uses recommendations. Also threw it in a sample to test and it gives me recommendations back for a dossier so I think it's all good

Reply:
The prefix denotes the purpose. This API doesn't set an object as recommended, it GETs a list of recommendations based on the query. From my perspective "recommend" seems a bit like "setRecommended" - thought "getRecommendations" would be more intuitive & self documenting from the fn name alone.

Reply:
Ah I see what you were saying. Thought you meant the API endpoint, yeah that name makes sense, i've updated it